commons-core/src/main/scala/com/avsystem/commons/serialization/cbor/RawCbor.scala

```diff
@@ -1,17 +1,42 @@
 package com.avsystem.commons
 package serialization.cbor
 
-import com.avsystem.commons.serialization.{GenCodec, TypeMarker}
+import com.avsystem.commons.serialization.{GenCodec, SizePolicy, TypeMarker}
+
+import scala.annotation.tailrec
 
 final case class RawCbor(bytes: Array[Byte], offset: Int, length: Int) {
   require(offset >= 0 && length >= 0 && offset + length <= bytes.length)
 
+  override def equals(other: Any): Boolean = other match {
+    case RawCbor(otherBytes, otherOffset, otherLength) =>
+      @tailrec def loop(i: Int): Boolean =
+        i >= length || bytes(offset + i) == otherBytes(otherOffset + i) && loop(i + 1)
```

Comment:
what about maps? Is it intentional that two equal maps before the serialization could in principle generate two non-equal byte arrays if the key ordering within a map is different?

Reply:
Is it a problem if you serialize the map to JSON string?

Reply:
Well, yes, it is related to JSON aswell. Do you think this is not worth bothering?

Reply:
I think it's not because it's a fundamental "problem" for every serialization.